[DDP] Remove the duplicate parseHookResult in reducer

wayi1 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #59523 [DDP] Rename the member divFactor_ as div_factor for naming consistency in reducer
* **#59510 [DDP] Remove the duplicate parseHookResult in reducer**
* #59576 [Reland][Gradient Compression] Apply division first to avoid overflow
* #59574 [Reland2][DDP] Merge work and future_work in reducer

Address the comment in https://github.com/pytorch/pytorch/pull/58937#discussion_r645822768

#Closes: https://github.com/pytorch/pytorch/issues/41266

Differential Revision: [D28918694](https://our.internmc.facebook.com/intern/diff/D28918694/)

Commit:
[DDP] Remove the duplicate parseHookResult in reducer

Address the comment in https://github.com/pytorch/pytorch/pull/58937#discussion_r645822768

#Closes: https://github.com/pytorch/pytorch/issues/41266

Differential Revision: [D28918694](https://our.internmc.facebook.com/intern/diff/D28918694/)

[ghstack-poisoned]

Commit:
Update on "[DDP] Remove the duplicate parseHookResult in reducer"


Address the comment in https://github.com/pytorch/pytorch/pull/58937#discussion_r645822768

#Closes: https://github.com/pytorch/pytorch/issues/41266

Differential Revision: [D28918694](https://our.internmc.facebook.com/intern/diff/D28918694/)

[ghstack-poisoned]

Commit:
Update on "[DDP] Remove the duplicate parseHookResult in reducer"


Address the comment in https://github.com/pytorch/pytorch/pull/58937#discussion_r645822768

#Closes: https://github.com/pytorch/pytorch/issues/41266

Differential Revision: [D28918694](https://our.internmc.facebook.com/intern/diff/D28918694/)

[ghstack-poisoned]

Commit:
Update on "[DDP] Remove the duplicate parseHookResult in reducer"


Address the comment in https://github.com/pytorch/pytorch/pull/58937#discussion_r645822768

#Closes: https://github.com/pytorch/pytorch/issues/41266

Differential Revision: [D28918694](https://our.internmc.facebook.com/intern/diff/D28918694/)

[ghstack-poisoned]

cbalioglu commented on torch/lib/c10d/comm.hpp:
```diff
@@ -87,6 +87,20 @@ class TORCH_PYTHON_API CommHookInterface {
 
+// This helper function is called both by CppCommHookInterface below and inside
+// reducer.
+inline std::vector<at::Tensor> _parseCppCommHookResult(const c10::IValue& result) {
```
What is the reason for prefixing this function name with an underscore? To mark it as an "internal" function? If that is the case, I suggest moving it to a `detail` namespace instead (common convention in C++).

```c++
namespace detail {
std::vector<at::Tensor> parseCppCommHookResult(const c10::IValue& result) {
  ...
}
}
```

SciPioneer replied:
Sounds good. I was wonder what's the good programming style in this codebase. It seems that creating a `details` namespace is consisting with other similar scenarios under `torch`.

Commit:
Update on "[DDP] Remove the duplicate parseHookResult in reducer"


Address the comment in https://github.com/pytorch/pytorch/pull/58937#discussion_r645822768

#Closes: https://github.com/pytorch/pytorch/issues/41266

Differential Revision: [D28918694](https://our.internmc.facebook.com/intern/diff/D28918694/)

[ghstack-poisoned]

cbalioglu reviewed (approved):
LGTM
